packages/sol-cov/src/artifact_adapters/truffle_artifact_adapter.ts

```diff
@@ -31,9 +46,38 @@ export class TruffleArtifactAdapter extends AbstractArtifactAdapter {
         };
         const compiler = new Compiler(compilerOptions);
         await compiler.compileAsync();
-        const solCompilerArtifactAdapter = new SolCompilerArtifactAdapter(artifactsDir, this._sourcesPath);
+        const solCompilerArtifactAdapter = new SolCompilerArtifactAdapter(artifactsDir, contractsDir);
         const contractsDataFrom0xArtifacts = await solCompilerArtifactAdapter.collectContractsDataAsync();
-        rimraf.sync(artifactsDir);
         return contractsDataFrom0xArtifacts;
     }
+    private _getTruffleConfig(): TruffleConfig {
+        const truffleConfigFileShort = path.resolve(path.join(this._projectRoot, 'truffle.js'));
+        const truffleConfigFileLong = path.resolve(path.join(this._projectRoot, 'truffle-config.js'));
+        if (fs.existsSync(truffleConfigFileShort)) {
+            const truffleConfig = require(truffleConfigFileShort);
+            return truffleConfig;
+        } else if (fs.existsSync(truffleConfigFileLong)) {
+            const truffleConfig = require(truffleConfigFileLong);
+            return truffleConfig;
+        } else {
+            throw new Error(
+                `Neither ${truffleConfigFileShort} nor ${truffleConfigFileLong} exists. Make sure the project root is correct`,
+            );
+        }
+    }
+    private _assertSolidityVersionIsCorrect(truffleArtifactsDirectory: string): void {
+        const artifactsGlob = `${truffleArtifactsDirectory}/**/*.json`;
+        const artifactFileNames = glob.sync(artifactsGlob, { absolute: true });
+        for (const artifactFileName of artifactFileNames) {
+            const artifact = JSON.parse(fs.readFileSync(artifactFileName).toString());
+            const compilerVersion = artifact.compiler.version;
+            if (!compilerVersion.startsWith(this._solcVersion)) {
+                throw new Error(
+                    `${artifact.contractName} was compiled with solidity ${compilerVersion} but specified version is ${
+                        this._solcVersion
+                    } it will make sol-cov unable to process traces. Please pass the correct version`,
```

Comment:
`it will make sol-cov unable to process traces` -> `making it impossible for sol-cov to process traces`